src/helper/sampler.hpp

```diff
@@ -0,0 +1,84 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include <iostream>
+#include <mutex>
+
+#pragma once
+
+namespace dds {
+class sampler {
+public:
+    sampler(bool enabled, double sample_rate) : enabled_(enabled)
+    {
+        if (sample_rate < 0 || sample_rate > 1) {
+            sample_rate = default_sample_rate;
+        }
+
+        tokens_per_hundred_ = (unsigned)round(sample_rate * 100);
```

Comment:
100 could be a member variable, it would make it easier to change the magnitude, if you don't want to dynamically change it then it should be a const.

Reply:
Done